### ISSUE TITLE
PLO extension

### ISSUE THREAD

**Issue by TomKan0909** [b-inary/postflop-solver#23]
Hi @b-inary big fan of your work and just wanted to say thank you for open sourcing it. 

Is it possible to extend this library to solve PLO and PLO-5

Thanks

**Comment by b-inary** [comment]
Thank you for the compliment!

The solver engine itself is designed to be versatile, so extending it to PLOs is technically possible, but the required memory usage would be too large and would not work in practice. Existing solvers typically use a process called "abstraction" to reduce memory usage by treating hands of similar strength as the same hand. To support PLOs, a naive implementation will not suffice, and we will have to make some efforts like this.

Also, providing a nice GUI is a difficult problem. For these reasons, while I do understand the need for PLOs, I am sorry to say that supporting PLOs is a lower priority.

**Comment by TomKan0909** [comment]
Thanks for getting back to me so quickly with the detailed response! Unfortunate to hear that PLO is computationally too expensive  😢. Would have loved to tried an OSS PLO solver haha. Once again thank you for your OSS contribution

**Comment by bkushigian** [comment]
@b-inary I've been thinking about playing with some abstraction logic (I've been toying with some clustering algos in my spare time). I know you've expressed some hesitation in the past about using abstractions (in particular for implementing a preflop solver), but if I do get around to getting clustering stuff implemented would you be interested in integrating this?

**Comment by b-inary** [comment]
Of course, it would be great if we could create an open-source solver with a decent abstraction technique. The abstraction technique allows us to solve multi-way situations, preflop, Omaha, etc. This is what MonkerSolver is good at.

However, in my opinion, such a solver should have a different repository than this one. I think this repository should focus on computing an exact solution to the two-player situation. So the short answer is no; instead of integrating an abstraction implementation into this repository, I think we should create a new repository.
